assets/schema_fasta.json

```diff
@@ -0,0 +1,129 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "https://raw.githubusercontent.com/nf-core/eager/master/assets/schema_fasta.json",
+    "title": "nf-core/eager pipeline - params.fasta schema",
+    "description": "Schema for the file provided with params.fasta",
+    "type": "array",
+    "items": {
+        "type": "object",
+        "properties": {
+            "reference_name": {
+                "type": "string",
+                "pattern": "^\\S+$",
+                "meta": ["id"],
+                "unique": true,
+                "errorMessage": "Reference name must be provided and cannot contain spaces."
+            },
+            "fasta": {
+                "type": "string",
+                "format": "file-path",
+                "pattern": "^\\S+\\.f(na|asta|a|as)(\\.gz)?$",
+                "exists": true,
+                "unique": true,
+                "errorMessage": "Fasta files for the mapping reference must be provided with file extensions '.fasta', '.fa', '.fas', '.fna', '.fasta.gz','.fa.gz','.fas.gz', '.fna.gz' and cannot contain any spaces."
+            },
+            "fai": {
+                "type": "string",
+                "format": "file-path",
+                "pattern": "^\\S+\\.fai$",
+                "exists": true,
+                "errorMessage": "Fasta index files for the mapping reference cannot have any spaces and must have file extension '.fai'."
+            },
+            "dict": {
+                "type": "string",
+                "format": "file-path",
+                "pattern": "^\\S+\\.dict$",
+                "exists": true,
+                "errorMessage": "Picard sequence dictionary files for the mapping reference cannot have any spaces and must have file extensions '.dict'."
+            },
+            "mapper_index": {
+                "type": "string",
+                "format": "directory-path",
+                "pattern": "^\\S+$",
+                "exists": true,
+                "errorMessage": "The directories of the index files for the mapping reference for a given mapper must not contain any spaces and have file extensions ''."
+            },
+            "circular_target": {
+                "type": "string",
+                "pattern": "^\\S+$",
```

Comment:
Does this not complain if the parameter is not provided/is empty?

Reply:
no